src/server/replica.h

```diff
@@ -240,11 +151,97 @@ class Replica {
  private:
   Service& service_;
   MasterContext master_context_;
-  std::unique_ptr<util::LinuxSocketBase> sock_;
-  Mutex sock_mu_;
 
+  // In redis replication mode.
+  Fiber sync_fb_;
+  Fiber acks_fb_;
+  EventCount waker_;
+
+  std::vector<std::unique_ptr<DflyShardReplica>> shard_flows_;
   std::shared_ptr<MultiShardExecution> multi_shard_exe_;
 
+  // Guard operations where flows might be in a mixed state (transition/setup)
+  Mutex flows_op_mu_;
+
+  // repl_offs - till what offset we've already read from the master.
+  // ack_offs_ last acknowledged offset.
+  size_t repl_offs_ = 0, ack_offs_ = 0;
+  std::atomic<unsigned> state_mask_ = 0;
+  unsigned num_df_flows_ = 0;
+
+  bool is_paused_ = false;
+  std::string id_;
+};
+
+struct DflyShardReplica : public ProtocolClient {
```

Comment:
Can you please add a comment above this class and `Replica`, saying what they do and how they differ?